### ISSUE TITLE
How to solve the numeric error?

### ISSUE THREAD

**Issue by acgtyrant** [vacancy/Synchronized-BatchNorm-PyTorch#12]
What is the numeric error? Overflow? Or not same behavior as the original batchnorm? And could you give some hints to solve it? Thank you very much!

**Comment by vacancy** [comment]
"It is less numerically stable compared to the original PyTorch implementation."

Please see the code `tests/test_numeric_batchnorm.py` for more details. Comparing the handy implementation of BatchNorm and the PyTorch's BatchNorm, you can see a small numerical difference in the output/grad.
